### PR TITLE
Fix ResNet50 max-pooling padding

### PR DESCRIPTION
**Pull request by ndryden**
The max-pooling layer in ResNet should have padding 1.

It looks like the [original Caffe implementation](https://github.com/KaimingHe/deep-residual-networks) used 0 padding, but more recent implementations ([TensorFlow](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/examples/learn/resnet.py), [PyTorch](https://github.com/pytorch/vision/blob/master/torchvision/models/resnet.py)) use padding 1.

This also avoids shrinking the image unnecessarily, and avoids going from even to odd dimensions.

### CODE REVIEW

**Comment by timmoon10** [comment]
Does this change the dimension fed into the mean pooling layer at the end?

**Comment by ndryden** [comment]
It should still be 7x7.
